consensus/spos/bls/subroundBlock.go

```diff
@@ -314,15 +314,27 @@ func (sr *subroundBlock) receivedBlockBodyAndHeader(cnsDta *consensus.Message) b
 		return false
 	}
 
-	if sr.IsBlockBodyAlreadyReceived() {
+	if !sr.IsNodeLeaderInCurrentRound(node) { // is NOT this node leader in current round?
+		sr.ConsensusRating().Decrease(
```

Comment:
what will happen in case of forks ? decrease for the other leader which is in the other fork - even if my current node is in the wrong one ?

Reply:
Yes, but this is only in the local node view. It is no problem if that node will send afterward signatures and so on, it honesty will increase in the eyes of this node. Anyway we can set a generous threshold just to be sure that one node will not be banned too easy/soon. The same problem will happen also for nodes which come later in network and will do sync. They will don't now about these kind of local states like: how many times shards were stuck, if there was flooding in network, how many peers have been blacklisted and so on. Anyway I have added a round as a parameter in the call of method Increase and Decrease, just to have a possibility if we want to be more accurate, to inject this interface in BlockProcessor and on rollbacks we can remove the Increase/Decrease actions made in that rollback header round if there exist one of them.